packages/components/src/focal-point-picker/index.js

```diff
@@ -23,317 +26,170 @@ import {
-		event.preventDefault();
-		const value = this.getValueFromPoint(
-			{ x: event.pageX, y: event.pageY },
-			event.shiftKey
-		);
-		this.updateValue( value );
-		this.props.onDrag?.( value, event );
-	}
-	getValueFromPoint( point, byTenths ) {
-		const { bounds } = this.state;
-
-		const pickerDimensions = this.pickerDimensions();
-		const relativePoint = {
-			left: point.x - pickerDimensions.left,
-			top: point.y - pickerDimensions.top,
-		};
-
-		const left = Math.max(
-			bounds.left,
-			Math.min( relativePoint.left, bounds.right )
-		);
-		const top = Math.max(
-			bounds.top,
-			Math.min( relativePoint.top, bounds.bottom )
+const arrowKeyList = [ 'ArrowDown', 'ArrowLeft', 'ArrowRight', 'ArrowUp' ];
+
+export default function FocalPointPicker( {
+	autoPlay = true,
+	className,
+	help,
+	label,
+	onChange,
+	onDrag,
+	onDragEnd,
+	onDragStart,
+	resolvePoint,
+	url,
+	value: valueProp = {
+		x: 0.5,
+		y: 0.5,
+	},
+} ) {
+	const [ point, setPoint ] = useState( valueProp );
+	// Tracks the internal point value when it’s yet to be propagated.
+	const refPoint = useRef();
+	const keepPoint = ( value ) => {
+		setPoint( value );
+		refPoint.current = value;
+	};
+	const sendPoint = ( value = refPoint.current ) => {
+		onChange?.( value );
+		refPoint.current = undefined;
+	};
+	// Uses the internal point if it is held or else the value from props.
+	const { x, y } = refPoint.current !== undefined ? point : valueProp;
+
+	const dragAreaRef = useRef();
+	const [ bounds, setBounds ] = useState( INITIAL_BOUNDS );
+	const refUpdateBounds = useRef( () => {
+		const { clientWidth: width, clientHeight: height } =
+			dragAreaRef.current;
+		// Falls back to initial bounds if the ref has no size. Since styles
+		// give the drag area dimensions even when the media has not loaded
+		// this should only happen in unit tests (jsdom).
+		setBounds(
+			width > 0 && height > 0 ? { width, height } : { ...INITIAL_BOUNDS }
 		);
-
-		let nextX =
-			( left - bounds.left ) /
-			( pickerDimensions.width - bounds.left * 2 );
-		let nextY =
-			( top - bounds.top ) / ( pickerDimensions.height - bounds.top * 2 );
-
-		// Enables holding shift to jump values by 10%
-		const step = byTenths ? 0.1 : 0.01;
-
-		nextX = roundClamp( nextX, 0, 1, step );
-		nextY = roundClamp( nextY, 0, 1, step );
-
-		return { x: nextX, y: nextY };
-	}
-	pickerDimensions() {
-		const containerNode = this.containerRef.current;
-
-		if ( ! containerNode ) {
-			return {
-				width: 0,
-				height: 0,
-				left: 0,
-				top: 0,
-			};
+	} );
+
+	useEffect( () => {
+		const updateBounds = refUpdateBounds.current;
+		const { defaultView } = dragAreaRef.current.ownerDocument;
+		defaultView.addEventListener( 'resize', updateBounds );
+		return () => defaultView.removeEventListener( 'resize', updateBounds );
+	}, [] );
+
+	// Updates the bounds to cover cases of unspecified media or load failures.
+	useIsomorphicLayoutEffect( () => void refUpdateBounds.current(), [] );
```

Comment:
Is there any reason for discarding the `if (!url)` check in the last commit?

Reply:
When there is a `url` but it fails to load. It was what caused one of the regressions you noted:
>the thirds grid in the "Video" example does not adapt to its container